eosc/cmd/txCreate.go

```diff
@@ -26,14 +32,45 @@ var txCreateCmd = &cobra.Command{
 		actionBinary, err := api.ABIJSONToBin(contract, eos.Name(action), dump)
 		errorCheck("unable to retrieve action binary from JSON via API", err)
 
-		pushEOSCActions(api, &eos.Action{
-			Account:    contract,
-			Name:       action,
-			ActionData: eos.NewActionDataFromHexData([]byte(actionBinary)),
-		})
+		actions := []*eos.Action{
+			&eos.Action{
+				Account:    contract,
+				Name:       action,
+				ActionData: eos.NewActionDataFromHexData([]byte(actionBinary)),
+			}}
+
+		var contextFreeActions []*eos.Action
+		if forceUnique {
+			contextFreeActions = append(contextFreeActions, newNonceAction())
+		}
+
+		pushEOSCActionsAndContextFeeeActions(api, contextFreeActions, actions)
 	},
 }
 
+func newNonceAction() *eos.Action {
+	return &eos.Action{
+		Account: eos.AN("eosio.null"),
+		Name:    eos.ActN("nonce"),
+		ActionData: eos.NewActionData(system.Nonce{
+			Value: fmt.Sprintf("%d", time.Now().UnixNano()),
```

Comment:
C'est pas strictement garanti d'être un nonce / unique - tu pourrais avoir 2 processes / threads / 2 machines qui produisent la même trx avec le meme Unix timestamp en nanosec .... la probabilité est basse, mais pas nulle.

Je sais que `cleos` fait essentiellement ça ... mais c'est pas une raison pour faire comme eux.

Les nanosecondes donne `- au mieux - 10^9` d'entropie (en pratique moins pcq NTP essaye de réduire ça le plus possible by-design). `10^9` --> environ 32 bits d'entropie. Je crois que 6 bytes de pure random (48 bits) devrait être suffisant pour ce scénario.

Je propose:
```
nonce := make([]byte, 6) // 48 bits entropy
if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
  panic(err.Error())
}
return &eos.Action{
		Account: eos.AN("eosio.null"),
		Name:    eos.ActN("nonce"),
		ActionData: eos.NewActionData(system.Nonce{
			Value: hex.EncodeToString(nonce),
...
```

Reply:
100% d'accord avec toi, je vais faire les modifications. Let's do it better!